File: src/main/java/bme/creditcalc/ui/SemesterTableCellRenderer.java
package bme.creditcalc.ui;

import bme.creditcalc.Main;
import com.github.weisj.darklaf.LafManager;

import javax.swing.*;
import javax.swing.table.DefaultTableCellRenderer;
import javax.swing.table.TableCellRenderer;
import java.awt.*;

public class SemesterTableCellRenderer implements TableCellRenderer {
    TableCellRenderer defaultRenderer;
    public SemesterTableCellRenderer(TableCellRenderer defaultRenderer){
        this.defaultRenderer = defaultRenderer;
    }
    public Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus, int row, int column) {
        Component c = defaultRenderer.getTableCellRendererComponent(table, value, isSelected, hasFocus, row, column);
        if(((SemesterTable)table.getModel()).getSemester().getSubject(row).isFinalized()){
            c.setBackground(Window.DONE_COLOR);
        } else {
            c.setBackground(table.getBackground());
        }
        return c;
    }
}

File: src/main/java/bme/creditcalc/ui/SemesterTableCellRenderer.kt
package bme.creditcalc.ui

import java.awt.Component
import javax.swing.JTable
import javax.swing.table.TableCellRenderer

class SemesterTableCellRenderer(private var defaultRenderer: TableCellRenderer) : TableCellRenderer {
    override fun getTableCellRendererComponent(table: JTable, value: Any, isSelected: Boolean, hasFocus: Boolean, row: Int, column: Int): Component {
        val c = defaultRenderer.getTableCellRendererComponent(table, value, isSelected, hasFocus, row, column)
        if ((table.model as SemesterTable).semester!!.subjects[row].isFinalized) {
            c.background = Window.DONE_COLOR
        } else {
            c.background = table.background
        }
        return c
    }
}